frameworks/helloworld/tests/test_sidecar.py

```diff
@@ -49,8 +50,30 @@ def test_sidecar():
 def test_sidecar_parameterized():
     run_plan('sidecar-parameterized', {'PLAN_PARAMETER': 'parameterized'})
 
+@pytest.mark.sanity
+def test_toxic_sidecar_doesnt_trigger_recovery():
+    # 1. Run the toxic sidecar plan that will never succeed.
+    # 2. Restart the scheduler.
+    # 3. Verify that its recovery plan is empty, as a failed FINISHED task should
+    # never trigger recovery
+    recovery_plan = sdk_plan.get_plan(PACKAGE_NAME, 'recovery')
+    assert(len(recovery_plan['phases']) == 0)
+    sdk_utils.out(recovery_plan)
+    run_plan('sidecar-toxic', wait_for_completion=False)
+    # Wait for the bad sidecar plan to be starting.
+    sdk_plan.wait_for_starting_plan(PACKAGE_NAME, 'sidecar-toxic')
+
+    # Restart the scheduler and wait for it to come up.
+    sdk_marathon.restart_app(PACKAGE_NAME)
+    sdk_plan.wait_for_completed_deployment(PACKAGE_NAME)
+
+    # Now, verify that its recovery plan is empty.
+    sdk_plan.wait_for_completed_plan(PACKAGE_NAME, 'recovery')
+    recovery_plan = sdk_plan.get_plan(PACKAGE_NAME, 'recovery')
+    assert(len(recovery_plan['phases']) == 0)
+
 
-def run_plan(plan_name, params=None):
+def run_plan(plan_name, wait_for_completion=True, params=None):
```

Comment:
We could also have a `start_plan` method that we call on https://github.com/mesosphere/dcos-commons/pull/1333/files#diff-4668ccf67c53972f707957cb942c92e2R62 and call that from `run_plan`. That would remove the need for the `wait_for_completion` argument. Note that this addition here breaks the call to `run_plan` on https://github.com/mesosphere/dcos-commons/pull/1333/files#diff-4668ccf67c53972f707957cb942c92e2R51